src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserRoleService.cs

```diff
@@ -428,6 +429,57 @@ await _usersAndRolesPersistenceHelper
                 });
         }
 
+        public async Task<Either<ActionResult, Unit>> RemoveUserPreReleaseRoles(Guid userId)
+        {
+            return await _userService.CheckCanManageAllUsers()
+                .OnSuccess(async _ =>
+                {
+                    return await _contentPersistenceHelper
+                        .CheckEntityExists<User>(userId)
+                        .OnSuccess(async _ =>
+                        {
+
+                            var userReleaseRoles =
+                                await _contentDbContext.UserReleaseRoles.Where(urr => urr.UserId == userId)
```

Comment:
I'd say that the method name might be misleading here, or we're missing some checks in our code.

If the intention is to clear down a user's Prerelease Roles across the board: 

You'll need to add a check for `urr.Role == PreRelease` as well, as otherwise this'll clear down ALL of the user's ReleaseRoles for both prerelease and non-prerelease role types.

It'd be good to have a unit test that covers this scenario - i.e. set up a user with various ReleaseRoles, some of which are prerelease and some which are not, and then check that the non-prerelease ones have been correctly left alone.

Reply:
Or if the intention is to remove all Roles from a User, who at the current time is just a prerelease user:

I'd change the method name to `RemoveAllRolesFromPreReleaseUser`.

I'd then do a check to make sure that the user is legitimately only a Prerelease User, by ensuring that they have the Prerelease User global role.  If not, then this method should throw a wobbly.

Reply:
Throw a wobbly === `return ValidationActionResult(NameOfYourErrorMessageHere);`.

Have a search for "return ValidationActionResult" in the IDE and you'll see plenty of examples :+1:

Reply:
Cool, yeah I get why the method name is misleading. The pre-release users that the UI tests could theoretically 
become an analyst, bau, etc if someone ends up changing their role.